### PR TITLE
tom/mget_support

### PR DESCRIPTION
**Pull request by binarytom**

### CODE REVIEW

File: lib/Cache/RedisDB.pm [hunk 1]
```diff
@@ -75,26 +78,56 @@ sub redis {
...
+but it also accepts a list of C<@keys> to look up.
+
+Returns an arrayref in the same order as the original keys. For any
+key that had no value, the resulting arrayref will contain C<undef>.
+
+=cut
+
+sub mget {
+    my ($self, $namespace, @keys) = @_;
+    my $res = Cache::RedisDB::redis()->mget(map { Cache::RedisDB::_cache_key($namespace, $_) } @keys);
```
Comment (bork1n) [inline lib/Cache/RedisDB.pm]: in other places we use it as just `redis()`, without full name.

```perl
sub _decode {
my $data = shoft;
state $decoder = Sereal::Decoder->new();
return ($data && looks_like_sereal($data)) ? $decoder->decode($data) : $data
}
```
to a separate interval sub and use it from both `get` and `mget`?

Comment (binarytom) [reply lib/Cache/RedisDB.pm]: Good catch on the namespace prefix - will remove.

The extra sub call is unlikely to cause much noticeable overhead, so I'll make that change in this PR as well - thanks!

File: lib/Cache/RedisDB.pm [hunk 2]
```diff
@@ -75,26 +78,56 @@ sub redis {
...
+    state $decoder = Sereal::Decoder->new();
```
Comment (basiliscos) [inline lib/Cache/RedisDB.pm]: I see the similar line in `get`. May be unite them and use single decoder instance?

File: lib/Cache/RedisDB.pm [hunk 3]
```diff
@@ -75,26 +78,56 @@ sub redis {
...
+    return [ map {
+        ($_ && looks_like_sereal($_)) ? $decoder->decode($_) : $_
```
Comment (basiliscos) [inline lib/Cache/RedisDB.pm]: I'd suggest to warn, when it is defined, but does not `looks_like_serial`

Comment (binarytom) [reply lib/Cache/RedisDB.pm]: As far as I can tell, the intent in this module is to provide transparent Sereal support: any data that's Sereal-encoded should be decoded, other data is passed through as-is.

**Pull request merged**